mkosi/__init__.py

```diff
@@ -3464,29 +3469,33 @@ def make_extension_image(context: Context, output: Path) -> None:
         mounts += [Mount(context.config.verity_certificate, context.config.verity_certificate, ro=True)]
     if context.config.sector_size:
         cmdline += ["--sector-size", str(context.config.sector_size)]
-
-    env = {
-        option: value
-        for option, value in context.config.environment.items()
-        if option.startswith("SYSTEMD_REPART_MKFS_OPTIONS_") or option == "SOURCE_DATE_EPOCH"
-    }
+    if context.config.split_artifacts:
+        cmdline += ["--split=yes"]
 
     with complete_step(f"Building {context.config.output_format} extension image"):
-        r = context.resources / f"repart/definitions/{context.config.output_format}.repart.d"
-        mounts += [Mount(r, r, ro=True)]
-        run(
-            cmdline + ["--definitions", r],
-            env=env,
-            sandbox=context.sandbox(
-                binary="systemd-repart",
-                devices=(
-                    not context.config.repart_offline or
-                    context.config.verity_key_source.type != KeySource.Type.file
+        j = json.loads(
+            run(
+                cmdline,
+                stdout=subprocess.PIPE,
+                env=context.config.environment,
+                sandbox=context.sandbox(
+                    binary="systemd-repart",
+                    devices=(
+                        not context.config.repart_offline or
+                        context.config.verity_key_source.type != KeySource.Type.file
+                    ),
+                    mounts=mounts,
                 ),
-                mounts=mounts,
-            ),
+            ).stdout
         )
 
+    logging.debug(json.dumps(j, indent=4))
```

Comment:
It might be nice to not load the json right away, but only further down when it comes to getting the partition. This way here the actual repart output could be logged without round tripping it here.

Reply:
Ehh should be fine, let's do this in another PR as it needs to be done in make_image() as well